### PR TITLE
Add new property onSelect

### PR DESCRIPTION
**Pull request by ohyjek**
### Intro

The current functionality of this component does not allow for developers to handle exactly what happens with their own application state with each selection of a menu item. 

The current behavior is such that `onChange` is only called when the menu is closed. This PR addresses this issue by adding `onSelect`

### onSelect

onSelect is used to expose the selected values each time the user selects or unselects a value from the menu item. It differs from `onChange` because it is called _every_ time the user makes a new selection. 

This will be useful because developers now have access to the entire selection state inside the select field every time the user interacts with it. 

### Use case 

A developer is using the select field in a `dialog`. 

The developer would like for the dialog to be able to apply the selection even while the menu is open.

With this new property it's now possible to always know what's selected and hence even if the dialog holding the select field closes, we can save and make request on the items selected.

### Side note

Any current consumers of this component will be unaffected by this change. But allows those for who desire it a little more control over how their applications can respond to this component 💯 

Thoughts and feedback welcome 👍

### CODE REVIEW

File: src/SuperSelectField.js [hunk 1]
```diff
@@ -146,7 +151,7 @@ class SelectField extends Component {
       const updatedValues = selectedItemExists
         ? selectedItems.filter(obj => !areEqual(obj.value, selectedItem.value))
         : selectedItems.concat(selectedItem)
-      this.setState({ selectedItems: updatedValues })
+      this.setState({ selectedItems: updatedValues }, this.getSelected())
```
Comment (ohyjek) [inline src/SuperSelectField.js:154]: If for some reason the `setState` here fails, then `getSelected()` will not be called.

File: README.md [hunk 2]
```diff
@@ -56,7 +56,8 @@ This component requires 3 dependencies :
 | keepSearchOnSelect | bool | false | Prevents the autocomplete field's value to be reset after each selection.|
 | disabled | bool | false | Include this property to disable superSelectField.|
 | value | null, object, object[] | null | Selected value(s).<br>/!\ REQUIRED: each object must expose a 'value' property. |
-| onChange | function | () => {} | Triggers when selecting/unselecting an option from the Menu.<br>signature: (selectedValues, name) with `selectedValues` array of selected values based on selected nodes' value property, and `name` the value of the superSelectField instance's name property |
+| onChange | function | () => {} | Triggers when closing the menu. Use this if you do not want to update your component state with each selection and only on menu close. <br>signature: (selectedValues, name) with `selectedValues` array of selected values based on selected nodes' value property, and `name` the value of the superSelectField instance's name property |
```
Comment (ohyjek) [inline README.md:59]: I thought this might be more clear 🤷‍♂️

File: src/SuperSelectField.js [hunk 3]
```diff
@@ -78,6 +78,11 @@ class SelectField extends Component {
     })
   }
 
+  getSelected = () => {
+    const {onSelect, name} = this.props;
+    onSelection(this.state.selectedItems, name);
```
Comment (Sharlaan) [inline src/SuperSelectField.js]: typo onSelectION ?

Comment (ohyjek) [reply src/SuperSelectField.js]: Yes thank you :)

**Comment by ohyjek** [comment]
I am happy to write tests for this as well if that's required 👍

**Comment by Sharlaan** [comment]
Yes please some tests

**Comment by ohyjek** [comment]
Hi,

I am having some trouble trying to get sinon to work. I suspect I am missing a configuration option somewhere.

Here is my test

```js
  it('expects the onSelect handler to be called', () => {
    var callback = sinon.spy();
    const wrapper = shallowWithContext(<SuperSelectField multiple onSelect={callback}>{testChildren}</SuperSelectField>)
    wrapper.simulate('click')
    const firstChild = wrapper.find('ListItem').first()
    firstChild.simulate('onTouchTap')
    expect(callback.calledOnce).to.equal(true)
  })
```

And here is my error

```
FAILED TESTS:
  When selecting an option
    ✖ expects the onSelect handler to be called
      PhantomJS 2.1.1 (Mac OS X 0.0.0)
    undefined is not an object (evaluating '__WEBPACK_IMPORTED_MODULE_3_expect___default()(callback.calledOnce).to.equal')
    webpack:///src/SuperSelectField.test.js:103:0 <- src/SuperSelectField.test.js:96218:75
```
 
Do you have any thoughts @Sharlaan ?

**Pull request merged**